ddtrace/tracer/logger.go

```diff
@@ -0,0 +1,42 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package tracer
+
+import "github.com/DataDog/dd-trace-go/v2/internal/log"
+
+// Logger implementations are able to log given messages that the tracer or profiler might output.
```

Comment:
NIT: Mention AdeptLogger here or mark this as deprecated (via a comment) if you follow my `UseLevelLogger` idea below.